src/main/java/uk/ac/cam/cl/dtg/isaac/api/services/AssignmentService.java

```diff
@@ -39,24 +41,25 @@ public AssignmentService(final UserAccountManager userManager) {
         this.userManager = userManager;
     }
 
-    public <T extends IAssignmentLike> void augmentAssignerSummaries(Collection<T> assignments) throws SegueDatabaseException {
+    public <T extends IAssignmentLike> void augmentAssignerSummaries(final Collection<T> assignments) throws SegueDatabaseException {
         Map<Long, UserSummaryDTO> userSummaryCache = new HashMap<>();
 
+        // Iterating over the owner IDs allows us to cache "no user found" errors without querying database each time:
+        List<Long> ownerUserIds = assignments.stream().map(IAssignmentLike::getOwnerUserId).distinct().collect(Collectors.toList());
```

Comment:
Does `distinct()` take care of the `null`s?

Reply:
I'd not even considered that there might be `null` values. It shouldn't be possible for owner IDs to become null because there's a foreign key in Postgres with `ON DELETE CASCADE` (and we only soft-delete users anyhow now so a user ID can never vanish).

But I could add a `.filter(Objects::nonNull)` easily enough?

Reply:
Seeing there is a null check in the removed piece of code, I thought maybe that value was somehow nullable, or there was a NPE somewhere upstream.